notify/grafana_alertmanager.go

```diff
@@ -580,30 +580,35 @@ func TestTemplate(ctx context.Context, c TestTemplatesConfigBodyParams, tmpls []
 		}, nil
 	}
 
+	tc := templates.TemplateDefinition{
+		Name:     c.Name,
+		Template: c.Template,
+	}
+
 	// Recreate the current template replacing the definition blocks that are being tested. This is so that any blocks that were removed don't get defined.
 	var found bool
-	templateContents := make([]string, 0, len(tmpls)+1)
+	templateContents := make([]templates.TemplateDefinition, 0, len(tmpls)+1)
 	for _, td := range tmpls {
 		if td.Name == c.Name {
 			// Template already exists, test with the new definition replacing the old one.
-			templateContents = append(templateContents, c.Template)
+			templateContents = append(templateContents, tc)
 			found = true
 			continue
 		}
-		templateContents = append(templateContents, td.Template)
+		templateContents = append(templateContents, td)
 	}
 
 	if !found {
 		// Template is a new one, add it to the list.
-		templateContents = append(templateContents, c.Template)
+		templateContents = append(templateContents, tc)
 	}
 
 	// Capture the underlying text template so we can use ExecuteTemplate.
 	var newTextTmpl *tmpltext.Template
 	var captureTemplate template.Option = func(text *tmpltext.Template, _ *tmplhtml.Template) {
 		newTextTmpl = text
 	}
-	newTmpl, err := templateFromContent(templateContents, externalURL, captureTemplate)
+	newTmpl, err := templates.TemplateFromTemplateDefinitions(templateContents, log.NewNopLogger(), externalURL, captureTemplate)
```

Comment:
This function has a logger parameter already, no need to use `log.NewNopLogger()`